infoflow/entropy_kde.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+from infoflow import multidimkde as mdkde
+from math import log
+
+
+def kerneldensityestimation(source, target, timelagx, timelagy, n, bw_coeff):
+    """
+    :param source: source time series, 1-D np.ndarray
+    :param target: target time series, 1-D np.ndarray
+    :param timelagx: time lag in x from present. Integer.
+    :param timelagy: time lag in y from present. Integer.
+    :param n: number of equally spaced points along each dimension where probabilities are to be estimated. Integer.
+    :param bw_coeff: multiplier that adjusts Gaussian bandwidth. Integer.
+    :return: transfer entropy in bits. float.
+    """
+
+    """ All code written here is based on the transfer entropy
+     toolbox, Copyright 2011 Joon Lee. Use protected under GNU."""
+
+    # fixing block lengths to 1
+    l, k = 1, 1
+
+    sourcearr = source.flatten()
+    targetarr = target.flatten()
+
+    # populating source_pat, target_pat and target_t
+    source_pat = []
+    target_pat = []
+    target_t = []
+
+    for i in range(max(l+timelagx, k+timelagy)-1, min(len(sourcearr), len(targetarr))):
```

Comment:
You almost never want to loop over arrays in this fashion.

Read more about numpy array operations here: http://www.scipy-lectures.org/intro/numpy/index.html